config/retry/backoff.go

```diff
@@ -291,6 +291,25 @@ func (b *Backoffer) Fork() (*Backoffer, context.CancelFunc) {
 	}, cancel
 }
 
+// UpdateUsingForked updates Backoffer's status using forked one's
+// Note: Make sure forked is no longer used after this, because b reuses forked's errors/backoffSleepMS/backoffTimes fields
+func (b *Backoffer) UpdateUsingForked(forked *Backoffer) {
+	if forked == nil {
+		return
+	}
+	forkedParent := forked.parent
+	// We should check forkedParent == b here to only update bo with bo's child. However, in our existing code,
+	// we sometimes fork a already forked backoffer, so we need to update using the grand child backoffer.
+	if forkedParent == nil || (forkedParent != b && forkedParent.parent != b) {
```

Comment:
Better to check all ancestors for general purpose
```go
for bo := forked.parent; bo != nil; bo := bo.parent {
  if bo == b {
    b.totalSleep = forked.totalSleep
    b.excludedSleep = forked.excludedSleep
    b.errors = forked.errors
    b.backoffSleepMS = forked.backoffSleepMS
    b.backoffTimes = forked.backoffTimes
    break
  }
}
```

Reply:
Good idea, updated